src/App.tsx

```diff
@@ -15,71 +15,88 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  selectGood = (good: string) => {
+    this.setState({
+      selectedGood: good,
+    });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  clearSelection = () => {
+    this.setState({
+      selectedGood: '',
+    });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  renderGood = (good: string) => {
+    const isSelected = good === this.state.selectedGood;
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+    return (
+      <tr
+        key={good}
+        data-cy="Good"
+        className={isSelected ? 'has-background-scuccess-light' : ''}
+      >
+        <td>
+          {isSelected ? (
             <button
               data-cy="RemoveButton"
               type="button"
               className="button is-info"
+              onClick={this.clearSelection}
             >
               -
             </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
+          ) : (
             <button
               data-cy="AddButton"
               type="button"
               className="button"
+              onClick={() => this.selectGood(good)}
```

Comment:
also quite cool to save the comparison or the checking of the selected good into a variable!